src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestParticleWaypoint.kt

```diff
@@ -118,13 +121,21 @@ object PestParticleWaypoint {
             return
         }
         val waypoint = guessPosition ?: return
-        val color = LorenzColor.RED.toColor()
+        val distance = waypoint.distance(event.exactPlayerEyeLocation())
+        val color: Color
+        val middlePoint = GardenPlotApi.closestCenterPlot(waypoint)
+        if (middlePoint != null && waypoint.ceil().equalsIgnoreY(middlePoint)) {
```

Comment:
make this "is middle" check only once when the current guess position changes, not every render frame.